File: vue-frontend/src/models/game-data.ts
import {FrameElement} from "@/models/frame-element";
import {Events} from "@/models/events";
import {Player} from "@/models/player";
import {BasicGameInformation} from "@/models/basic-game-information";

export class GameData {
    id: string;
    basicInformation: BasicGameInformation;
    trackingAway: FrameElement[];
    trackingHome: FrameElement[];
    playersAway: Map<string, Player>;
    playersHome: Map<string, Player>;
    traveledDistancesAway: Map<string, []>;
    traveledDistancesHome: Map<string, []>;
    events: Events;
    asJson: string;

    constructor(init: any) {
        this.id = init?.id;
        this.basicInformation = init?.basicInformation;
        this.trackingAway = init?.trackingAway;
        this.trackingHome = init?.trackingHome;
        this.playersAway = init?.playersAway;
        this.playersHome = init?.playersHome;
        this.traveledDistancesAway = init?.traveledDistancesAway;
        this.traveledDistancesHome = init?.traveledDistancesHome;
        this.events = init?.events;
        this.asJson = init?.asJson;
    }
}

File: python-backend/models/gameData.py
import json
from typing import List
from models.frameElement import *
from models.events import Events
from models.player import Player


class GameData:
    def __init__(self):
        self.id: str
        self.basicInformation: dict[str, any]
        self.trackingHome: List[FrameElement] = []
        self.trackingAway: List[FrameElement] = []
        self.traveledDistancesHome: dict[str, dict]
        self.traveledDistancesAway: dict[str, dict]
        self.playersHome: dict[str, Player]
        self.playersAway: dict[str, Player]
        self.events: Events
        self.asJson: str = None

    def toJSON(self):
        if self.asJson:
            return self.asJson
        self.asJson = json.dumps(self, default=lambda o: o.__dict__)
        return self.asJson
